dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/BeamSQLFnExecutor.java

```diff
@@ -74,7 +79,7 @@ public BeamSQLFnExecutor(BeamRelNode relNode) {
    * {@link #buildExpression(RexNode)} visits the operands of {@link RexNode} recursively,
    * and represent each {@link SqlOperator} with a corresponding {@link BeamSqlExpression}.
    */
-  private BeamSqlExpression buildExpression(RexNode rexNode) {
+  static BeamSqlExpression buildExpression(RexNode rexNode) {
```

Comment:
just realised that '+' also support DATE, since DATE type support has not been merged, I will not support it in the PR.  Created an issue to track this: https://issues.apache.org/jira/browse/BEAM-2203